deprecation/middleware.go

```diff
@@ -0,0 +1,75 @@
+package deprecation
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+)
+
+type Deprecation struct {
+	Paths   []string
+	Date    string
+	Link    string
+	Message string
+	Sunset  string
+	Outages []Outage
+}
+
+type Outage struct {
+	Start time.Time
+	End   time.Time
+}
+
+func DeprecationRouter(deprecations []Deprecation) func(http.Handler) http.Handler {
```

Comment:
Ideally the function name shouldn't start with the package name
```suggestion
func DeprcRouter(deprecations []Deprecation) func(http.Handler) http.Handler {
```

Reply:
Fixed by removing `Deprecation` altogether rather than truncating it.

Reply:
good idea